src/main/java/org/openstreetmap/atlas/geography/atlas/BareAtlas.java

```diff
@@ -560,9 +568,24 @@ public SnappedEdge snapped(final Location point, final Distance threshold)
         nodesWithin(boundary, hasNode.negate()).forEach(
                 node -> builder.addNode(node.getIdentifier(), node.getLocation(), node.getTags()));
 
-        // Add the edges
-        edgesIntersecting(boundary).forEach(
-                edge -> builder.addEdge(edge.getIdentifier(), edge.asPolyLine(), edge.getTags()));
+        // Add the edges. Use a consumer that makes sure master edges are always added first.
```

Comment:
So if the `1` and `-1` have not been added yet there are multiple use cases:
- The consumer is told to add `1` first.
  - If `1` has a reverse edge, it will add `-1` (if not added already), then `1` (if not added already)
  - If `1` does not have a reverse edge, it will just add `1` (if not added already).
- The consumer is told to add `-1` first.
  - It will add `1` first (if not added already) then `-1` (if not added already)

Reply:
So in sub-bullet point 1 under bullet point 1, you still add the reverse edge before the master edge. Is this OK in this case? I guess I am just confused by the discrepancy between the comment and what actually happens.

Also, I am curious why we need to make sure the master gets added first. Sorry to be probing, I just want to make sure I understand!

Reply:
Oh now I understand. The problem was that the `Iterable` returned by `edgesIntersecting` is sometimes missing the master edge. So it doesn't actually matter which is added first, just that we always add the reverse edge/master edge together if both exist.